### ISSUE TITLE
Flutter 2.5: Migrate to recommended `flutter_lints` package

### ISSUE THREAD

**Issue by manjotsidhu** [CircuitVerse/mobile-app#136]
## What is `flutter_lints` ?

The package:flutter_lints defines the latest set of recommended lints that encourage good coding practices for Flutter apps, packages, and plugins. Projects created with flutter create using Flutter version 2.3.0-12.0.pre or newer are already enabled to use the latest set of recommended lints. Projects created prior to that version can upgrade to it with the instructions in this guide.
https://flutter.dev/docs/release/breaking-changes/flutter-lints-package

## TL;DR

Flutter 2.5 introduced it's own package `flutter_lints` and is recommended now to use since 2.5.
Migration Guide: https://flutter.dev/docs/release/breaking-changes/flutter-lints-package#migration-guide

**Comment by Neha62-lit** [comment]
@manjotsidhu, In flutter_lints dependencies, there are lot of changes in the files(almost 151 files) like remove the deprecated code and some warnings etc. that I have to make in the code. So What can I do now?
Should I make all those changes and push the code?

**Comment by Neha62-lit** [comment]
Hey @manjotsidhu, 
Can you please tell me, What should I do now?

**Comment by manjotsidhu** [comment]
> Hey @manjotsidhu, Can you please tell me, What should I do now?

@Neha62-lit You can pull the changes and check now if most of  the warnings and errors are fixed. If not then kindly send screenshot of the warnings/errors you are getting.

**Comment by Neha62-lit** [comment]
Yes, Sure!

**Comment by Neha62-lit** [comment]
After running flutter analyze command, I am getting this warnings. Should I fix all those warnings or not?

![Screenshot (159)](https://user-images.githubusercontent.com/70088044/135743874-b434bbf6-b115-4243-ba2a-01474bcde4bd.png)

**Comment by manjotsidhu** [comment]
> After running flutter analyze command, I am getting this warnings. Should I fix all those warnings or not?
> 
> ![Screenshot (159)](https://user-images.githubusercontent.com/70088044/135743874-b434bbf6-b115-4243-ba2a-01474bcde4bd.png)

Would be great if you fix them since we are not following recommended set of lints. Though this issue was for migration to `flutter_lints` package, the scope of fixing those issues (468) are way too much for this migration and this can lead to a lot of code in a single PR. We can do the following steps -

- [ ] Create another branch, `fix-lints`. Target all changes for fixing analyzer issues to this branch.
- [ ] Pull Request for migration to `flutter_lints` package
- [ ] Submit one PR at a time for single type of linting issue. For example: group and fix lint "prefer const with constant constructors" everywhere in the codebase.
- [ ] Once all issues are fixed, we will merge `fix-lints` to `master` (without squash and merge)

I hope this sounds good ?
I will also be contributing on that same branch for linting issues and together we can fix 468 issues.

**Comment by Neha62-lit** [comment]
I guess Option1 would be great.

**Comment by manjotsidhu** [comment]
I feel a major drawback of Option 1 is that there will be new commits coming on `master` branch and later on we will have to fix merge conflicts but in Option 2 there are less chances of merge conflicts since every PR will target `master` branch.

**Comment by tachyons** [comment]
It depends on two factors
1. how much effort we need to put to fix linting issues 
2. How frequent are the updates to the master branch

Is the current master branch passing CI ?

**Comment by manjotsidhu** [comment]
> It depends on two factors
> 
> 1. how much effort we need to put to fix linting issues
> 2. How frequent are the updates to the master branch
> 
> Is the current master branch passing CI ?

1. 468 Analyzer issues may need significant collaborative efforts.
2. As you can see by the pull requests of contributions, we can say 1 PR/week.

The current branch of master is passing both CI and CD.

**Comment by manjotsidhu** [comment]
@Neha62-lit I think it would be easy for us to go for Option 1. You can make the PR for migration and also ignore all the issues that analyzer is throwing, example :
`analysis_options.yaml`
```
linter:
  rules:
    prefer_single_quotes: true  # Disable the `prefer_single_quotes` rule

# Additional information about this file can be found at
# https://dart.dev/guides/language/analysis-options
```

**Comment by Neha62-lit** [comment]
Okay
